optimum/habana/distributed/strategy.py

```diff
@@ -0,0 +1,112 @@
+from abc import abstractmethod
+from typing import Any, List, Mapping
+
+import torch
+import torch.distributed
+from torch import nn
+
+#from optimum.habana.distributed import tp_wrapping
+
+
+class DistributedStrategy:
+    def __init__(self, from_meta=False):
+        self.from_meta = from_meta
+
+    def distribute_module(
+        self, module: nn.Module, final_layers: bool = False
+    ) -> nn.Module:
+        """
+        Optionally a distributed strategy may distribute modules that are not
+        numbered layers
+        """
+        return module
+
+    @abstractmethod
+    def distribute_layer(self, block: nn.Module, layer: int) -> nn.Module:
+        """
+        Distribute each layer as-appropriate
+        """
+        pass
+
+
+class NotDistributed(DistributedStrategy):
```

Comment:
why derived class is NotDistributed for the base class as DistributedStrategy? It is creating some confusion in readability, may require some other name?